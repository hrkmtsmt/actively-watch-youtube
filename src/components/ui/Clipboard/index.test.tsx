import React from 'react';
import { describe, test, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Clipboard } from '.';

describe('Clipboard component', () => {
  test('snapshot copy icon', () => {
    render(<Clipboard label="Share" value="https://example.com" />);
    expect(document.body.getElementsByTagName('label')[0]).toMatchSnapshot();
  });

  test('snapshot success icon', () => {
    render(<Clipboard label="Share" value="https://example.com" isSuccess />);
    expect(document.body.getElementsByTagName('label')[0]).toMatchSnapshot();
  });
});
