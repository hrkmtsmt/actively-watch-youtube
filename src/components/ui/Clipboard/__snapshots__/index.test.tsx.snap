// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Clipboard component > snapshot copy icon 1`] = `
<label
  class="style_label__emdbfx0"
>
  <span
    class="style_name__emdbfx1"
  >
    Share
  </span>
  <div
    class="style_wrapper__f1rb9b0"
  >
    <input
      class="style_input__100qk3h2"
      disabled=""
      value="https://example.com"
    />
    <button
      class="style_button__1o6r5610"
      type="button"
    >
      <svg
        class="style_icon__1o6r5611"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="13"
          rx="2"
          ry="2"
          width="13"
          x="9"
          y="9"
        />
        <path
          d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
        />
      </svg>
    </button>
  </div>
</label>
`;

exports[`Clipboard component > snapshot success icon 1`] = `
<label
  class="style_label__emdbfx0"
>
  <span
    class="style_name__emdbfx1"
  >
    Share
  </span>
  <div
    class="style_wrapper__f1rb9b0"
  >
    <input
      class="style_input__100qk3h2"
      disabled=""
      value="https://example.com"
    />
    <button
      class="style_button__1o6r5610"
      type="button"
    >
      <svg
        class="style_icon__1o6r5611"
        fill="none"
        height="24"
        stroke="var(--colors-success__1b26nuq4)"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </button>
  </div>
</label>
`;
